streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -294,11 +306,13 @@ public synchronized void start() {
         log.debug("Starting Kafka Stream process");
 
         if (state == KafkaStreams.State.CREATED) {
-            for (final StreamThread thread : threads)
-                thread.start();
-
             setState(KafkaStreams.State.RUNNING);
 
+            for (final StreamThread thread : threads) {
```

Comment:
Any reason why you did revert the order? It seems more natural to me (but this is nit) to change the state after the threads got started, ie, (1) start thread (2) change thread state (3) after all threads got started change KS state.